### PR TITLE
Stabilize macro_literal_matcher

### LINKED ISSUE #35625: Tracking issue for `literal` fragment specifier (RFC 1576)
Tracking issue for rust-lang/rfcs#1576.

### PR DESCRIPTION
**Pull request by da-x**
This followed FCP in #35625.

Closes #35625

### CODE REVIEW

File: src/libsyntax/ext/tt/macro_rules.rs [hunk 1]
```diff
@@ -1027,26 +1027,21 @@ fn has_legal_fragment_specifier(sess: &ParseSess,
...
-                               features: &Features,
-                               attrs: &[ast::Attribute],
+fn is_legal_fragment_specifier(_sess: &ParseSess,
+                               _features: &Features,
+                               _attrs: &[ast::Attribute],
                                frag_name: &str,
-                               frag_span: Span) -> bool {
+                               _frag_span: Span) -> bool {
+    /*
+     * If new fragmnet specifiers are invented in nightly, `_sess`,
```
Comment (petrochenkov) [inline src/libsyntax/ext/tt/macro_rules.rs]: Typo: fragmnet -> fragment

File: src/test/run-pass/issues/issue-52169.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@
 // except according to those terms.
 
 // run-pass
-#![feature(macro_literal_matcher)]
+#![cfg_attr(stage0, feature(macro_literal_matcher))]
```
Comment (petrochenkov) [inline src/test/run-pass/issues/issue-52169.rs]: These are not necessary in tests, you can just remove the feature.

**Comment by petrochenkov** [comment]
r=me with nits addressed

**Comment by da-x** [comment]
Nits addressed, fixed another typo on the way.

r? @petrochenkov

**Comment by eddyb** [comment]
@da-x You have accidental submodule changes (you can use e.g. `git gui`'s amend mode to remove them from the commit).

**Comment by da-x** [comment]
@eddyb thanks. Amended the top commit.

**Comment by petrochenkov** [comment]
@da-x 
Could you remove the merge commit and rebase this patch on master instead?

**Comment by da-x** [comment]
@petrochenkov done

**Comment by da-x** [comment]
@petrochenkov, fixed again. Conflict was due to 57b7d55591d0a4a0d9aa16afa0f20

**Pull request merged**